src/posit/connect/permissions.py

```diff
@@ -67,36 +67,105 @@ def count(self) -> int:
         return len(self.find())
 
     @overload
-    def create(self, *, principal_guid: str, principal_type: str, role: str) -> Permission:
+    def create(self, *, principal_guid: str, principal_type: str, role: str) -> Permission: ...
+
+    @overload
+    def create(self, *args: User | Group, role: str) -> list[Permission]: ...
+
+    def create(self, *args: User | Group, **kwargs) -> Permission | list[Permission]:
         """Create a permission.
 
         Parameters
         ----------
+        *args : User | Group
+            The principal users or groups to add.
+        role : str
+            The principal role. Currently only `"viewer"` and `"owner"` are supported.
         principal_guid : str
+            User guid or Group guid.
         principal_type : str
+            The principal type. Either `"user"` or `"group"`.
         role : str
+            The principal role. Currently only `"viewer"` and `"owner"` are supported
 
         Returns
         -------
-        Permission
-        """
-
-    @overload
-    def create(self, **kwargs) -> Permission:
-        """Create a permission.
+        Permission | List[Permission]
+            Returns a `Permission` when the kwargs: `principal_guid`, `principal_type`, and `role`
+            are used. Returns a `list[Permission]` when `*args` are used.
```

Comment:
Returning either an object of a specific type or a list of those objects feels a little weird as an interface here. Is that a common approach for things like these? It certainly would make programming with the SDK more finicky because one would have to remember when it'll be a list and when it won't, yeah?

Reply:
The `@overload`s should handle the typing interface. 

If this is a road we don't want for the package, I'll need to back it out of the other PRs as well.

Reply:
Oooooo. I like our suggestion below that we keep the same return type of `Permission` but essentially `*args` turns into `principal: User | Group` as a position only argument.

Reply:
I've backed out implementations that "handle arrays of information because why not" in `User.groups.add(group)`, `Group.members.add(user)`, and `Permissions.destroy(permission)` in [`ab5a4d5` (#341)](https://github.com/posit-dev/posit-sdk-py/pull/341/commits/ab5a4d5364865e8740889ed8a6287859101993ec)

This way if its a single API call under the hood, things are more explicit and less magic. (No need to wrap the for loop and then insert confusion / misdirection.) If a day comes where the Connect API handles array inputs / results, then I'm up for supporting it within the SDK.